frontend/src/components/redirects/HomeRedirect.tsx

```diff
@@ -13,31 +13,49 @@ export function HomeRedirect() {
     if (!info) {
       return;
     }
+
+    const setupReturnTo = window.localStorage.getItem(
+      localStorageKeys.setupReturnTo
+    );
+
     let to: string | undefined;
-    if (info.setupCompleted && info.running) {
-      if (info.unlocked) {
-        if (info.albyAccountConnected || !info.albyUserIdentifier) {
-          const returnTo = window.localStorage.getItem(
-            localStorageKeys.returnTo
-          );
-          // setTimeout hack needed for React strict mode (in development)
-          // because the effect runs twice before the navigation occurs
-          setTimeout(() => {
-            window.localStorage.removeItem(localStorageKeys.returnTo);
-          }, 100);
-          to = returnTo || "/home";
+    if (!setupReturnTo) {
+      if (info.setupCompleted && info.running) {
+        if (info.unlocked) {
+          if (info.albyAccountConnected || !info.albyUserIdentifier) {
+            const returnTo = window.localStorage.getItem(
+              localStorageKeys.returnTo
+            );
+            // setTimeout hack needed for React strict mode (in development)
+            // because the effect runs twice before the navigation occurs
+            setTimeout(() => {
+              window.localStorage.removeItem(localStorageKeys.returnTo);
+            }, 100);
+            to = returnTo || "/home";
+          } else {
+            to = "/alby/auth";
+          }
         } else {
-          to = "/alby/auth";
+          to = "/unlock";
         }
+      } else if (info.setupCompleted && !info.running) {
+        to = "/start";
+      } else if (info.albyAccountConnected) {
+        to = "/welcome";
```

Comment:
If we are adding this, then we should also skip showing `ConnectAlbyAccount` screen if it is connected already. Can be done by adding a small `if (!info.albyAccountConnected)` check at the top in `navigateToAuthPage`

Reply:
Note: This particular fix was put in for if the user logs in, then navigates back. Because of how the OAuth redirect works, the back button actually takes them to `/`, which then redirects them to the `intro` again. It is possibly a bit of an edge case but I thought the UX was just so bad there (and is hard to fix due to the OAuth redirect).

Reply:
Done, and added comments to both

Reply:
🙌